loopdb/sqlite.go

```diff
@@ -369,20 +370,27 @@ func parseTimeParts(datePart, timePart string) (time.Time, error) {
 
 	year, err := strconv.Atoi(dateParts[0])
 	if err != nil {
-		return time.Time{}, err
+		return time.Time{}, fmt.Errorf("unable to parse year: %v", err)
 	}
 
 	month, err := strconv.Atoi(dateParts[1])
 	if err != nil {
-		return time.Time{}, err
+		return time.Time{}, fmt.Errorf("unable to parse month: %v", err)
 	}
 
 	day, err := strconv.Atoi(dateParts[2])
 	if err != nil {
-		return time.Time{}, err
+		return time.Time{}, fmt.Errorf("unable to parse day: %v", err)
 	}
 
 	// Parse the time.
+
+	// Remove the timezone part if it exists. E.g. 2023-08-04 16:07:49+0800 CST
+	if strings.Contains(timePart, "+") {
```

Comment:
I'm afraid there's some more things to consider. I was trying to look up where these things come from and it seems to me that this is governed by the ISO 8601 standard (https://en.wikipedia.org/wiki/ISO_8601). According to that specification the offset may also be negative.

Reply:
+1